src/PublicSuffixListManager.php

```diff
@@ -57,68 +48,82 @@ class PublicSuffixListManager
     /**
      * Public constructor.
      *
-     * @param HttpAdapter $httpAdapter
-     * @param string      $cacheDir    Optional cache directory
+     * @param CacheInterface $cacheAdapter
+     * @param HttpAdapter    $httpAdapter
+     * @param string         $sourceUrl
      */
-    public function __construct(HttpAdapter $httpAdapter, string $cacheDir = null)
-    {
-        $this->cacheDir = $cacheDir ?? realpath(dirname(__DIR__) . DIRECTORY_SEPARATOR . 'data');
+    public function __construct(
+        CacheInterface $cacheAdapter,
+        HttpAdapter $httpAdapter,
+        string $sourceUrl = self::PUBLIC_SUFFIX_LIST_URL
+    ) {
+        $this->cacheAdapter = $cacheAdapter;
         $this->httpAdapter = $httpAdapter;
+        $this->sourceUrl = $sourceUrl;
     }
 
     /**
      * Gets Public Suffix List.
      *
-     * @param string $list the Public Suffix List type
+     * @param string $type the Public Suffix List type
+     *
+     * @throws RuntimeException if no PublicSuffixList can be returned
      *
      * @return PublicSuffixList
      */
-    public function getList($list = self::ALL_DOMAINS): PublicSuffixList
+    public function getList($type = self::ALL_DOMAINS): PublicSuffixList
     {
-        $cacheBasename = isset(self::$domainList[$list]) ? self::$domainList[$list] : self::PDP_PSL_PHP_FILE;
-        $cacheFile = $this->cacheDir . '/' . $cacheBasename;
-        if (!file_exists($cacheFile)) {
-            $this->refreshPublicSuffixList();
+        static $type_lists = [self::ALL_DOMAINS => 1, self::ICANN_DOMAINS => 1, self::PRIVATE_DOMAINS => 1];
+
+        if (!isset($type_lists[$type])) {
```

Comment:
Since this is a new version should we fallback to the full public suffix list or just throw ? This fallback was interesting to avoid BC break if it was just a 3.x release ?